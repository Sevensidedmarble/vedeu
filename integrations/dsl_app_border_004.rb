#!/usr/bin/env ruby

# frozen_string_literal: true

require 'bundler/setup'
require 'vedeu'

class DSLApp

  Vedeu.bind(:_initialize_) { Vedeu.trigger(:_refresh_) }

  Vedeu.configure do
    debug!
    log '/tmp/vedeu_views_dsl.log'
    renderers [
                Vedeu::Renderers::Terminal.new(
                  filename:   '/tmp/dsl_app_border_004.out',
                  write_file: true),
                # Vedeu::Renderers::JSON.new(filename: '/tmp/dsl_app_border_004.json'),
                # Vedeu::Renderers::HTML.new(filename: '/tmp/dsl_app_border_004.html'),
                # Vedeu::Renderers::Text.new(filename: '/tmp/dsl_app_border_004.txt'),
              ]
    run_once!
    standalone!
  end

  load './support/test_interface.rb'

  Vedeu.border :test_interface do
    show_bottom false
  end

  Vedeu.render do
    view(:test_interface) do
      lines do
        line 'no bottom'
      end
    end
  end

  def self.actual
    File.read('/tmp/dsl_app_border_004.out')
  end

  def self.expected
    File.read(File.expand_path('../expected/dsl_app_border_004.out', __FILE__))
  end

  def self.start(argv = ARGV)
    Vedeu::Launcher.execute!(argv)
  end

end # DSLApp

DSLApp.start

if DSLApp.expected == DSLApp.actual
  puts "#{__FILE__} \e[32mPassed.\e[39m"
  exit 0;
else
  puts "#{__FILE__} \e[31mFailed.\e[39m"
  exit 1;
end
