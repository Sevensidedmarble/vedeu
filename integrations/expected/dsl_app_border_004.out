[0m[2;2H[39m[49m(0l(B[2;3H(0q(B[2;4H(0q(B[2;5H(0q(B[2;6H(0q(B[2;7H(0q(B[2;8H(0q(B[2;9H(0q(B[2;10H(0q(B[2;11H(0q(B[2;12H(0q(B[2;13H(0q(B[2;14H(0q(B[2;15H(0q(B[2;16H(0q(B[2;17H(0q(B[2;18H(0q(B[2;19H(0q(B[2;20H(0q(B[2;21H(0q(B[2;22H(0q(B[2;23H(0q(B[2;24H(0q(B[2;25H(0q(B[2;26H(0q(B[2;27H(0q(B[2;28H(0q(B[2;29H(0q(B[2;30H(0q(B[2;31H(0q(B[2;32H(0k(B[0m[3;2H[39m[49m(0x(B[3;3Hn[3;4Ho[3;5H [3;6Hb[3;7Ho[3;8Ht[3;9Ht[3;10Ho[3;11Hm[3;32H(0x(B[0m[4;2H[39m[49m(0x(B[4;32H(0x(B[0m[5;2H[39m[49m(0x(B[5;32H(0x(B[0m[6;2H[39m[49m(0x(B[6;32H(0x(B[0m